app/views/devise/registrations/new.html.erb

```diff
@@ -1,29 +1,46 @@
-<h2>Sign up</h2>
+<div class='grid-container'>
+  <div class='grid-x grid-margin-x grid-padding-y'>
+    <div class="cell auto">
 
-<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
-  <%= render "devise/shared/error_messages", resource: resource %>
+    </div>
+    <div class="cell small-12 medium-10">
+      <h2 class="text-white title">Sign up</h2>
 
-  <div class="field">
-    <%= f.label :email %><br />
-    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
-  </div>
+      <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+      <%= render "devise/shared/error_messages", resource: resource %>
 
-  <div class="field">
-    <%= f.label :password %>
-    <% if @minimum_password_length %>
-    <em>(<%= @minimum_password_length %> characters minimum)</em>
-    <% end %><br />
-    <%= f.password_field :password, autocomplete: "new-password" %>
-  </div>
+      <div class="field">
+        <%= f.label :email, class: ['text-white', 'body'] %><br />
+        <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+      </div>
 
-  <div class="field">
-    <%= f.label :password_confirmation %><br />
-    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
-  </div>
+      <div class="field">
+        <%= f.label :password, class: ['text-white', 'body']  %>
+        <% if @minimum_password_length %>
+        <em class="text-white body">(<%= @minimum_password_length %> characters minimum)</em>
+        <% end %><br />
+        <%= f.password_field :password, autocomplete: "new-password" %>
+      </div>
 
-  <div class="actions">
-    <%= f.submit "Sign up" %>
-  </div>
-<% end %>
+      <div class="field">
+        <%= f.label :password_confirmation, class: ['text-white', 'body']  %><br />
+        <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+      </div>
 
-<%= render "devise/shared/links" %>
+      <div class="field">
+        <%= f.label :profile_photo, class: ['text-white', 'body']  %>
+        <%= f.file_field :profile_photo, class: ['text-white', 'body']   %>
+      </div>
+
+      <div class="actions">
+        <%= f.submit "Sign up", class: 'button' %>
+        <%= render "devise/shared/links" %>
+      </div>
+      <% end %>
+
+    </div>
+    <div class="cell auto">
+
+    </div>
```

Comment:
What is this empty div doing?

Reply:
@safang99 there's one on each side of the content div to center it on the page

Reply:
@safang99 Eliminated the extra divs by using the appropriate Foundation class on the containing div.